src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -84,7 +84,7 @@
         (
             "numeric_literal",
             "regex",
-            r"([0-9]+(\.[0-9]+)?)|(\.[0-9]+)",
+            r"(\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?",
```

Comment:
I lifted this regex from [Stack Overflow](https://stackoverflow.com/questions/12929308/python-regular-expression-that-matches-floating-point-numbers/12929311).